openedx/core/djangoapps/notifications/serializers.py

```diff
@@ -26,3 +27,28 @@ class NotificationCourseEnrollmentSerializer(serializers.ModelSerializer):
     class Meta:
         model = CourseEnrollment
         fields = ('course', 'is_active', 'mode')
+
+
+class UserNotificationPreferenceSerializer(serializers.ModelSerializer):
+    """
+    Serializer for user notification preferences.
+    """
+    course_name = serializers.SerializerMethodField(read_only=True)
+
+    class Meta:
+        model = NotificationPreference
+        fields = ('id', 'course_name', 'course_id', 'notification_preference_config',)
+        read_only_fields = ('id', 'course_name', 'course_id',)
+        write_only_fields = ('notification_preference_config',)
+
+    def get_course_name(self, obj):
+        """
+        Returns course name from course id.
+        """
+        return CourseOverview.get_from_id(obj.course_id).display_name
+
+    def update(self, instance, validated_data):
```

Comment:
We might need a validation for notification_preference_config object , If API sends non existing preference, empty list ,  or list with random stuff what will be behaviour of API ?

Reply:
Yes, we need validation here. That will be done in a followup PR, once we have confirmed with FE what the request payload looks like.